components/EventForm/index.js

```diff
@@ -0,0 +1,101 @@
+import { StyledForm, StyledHeading, StyledLabel, StyledInput, StyledSaveButton, StyledCldUploadButton, StyledGoBackButton } from "./EventForm.styled";
+import StyledDatePicker from "./StyledDatePicker";
+import { useState } from "react";
+import Image from "next/image";
+import useSWR from "swr";
+
+
+export default function EventForm({ kidData }) {
+  const { mutate } = useSWR(`/api/kids/${kidData?._id}`);
+  
+  async function handleSubmit(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const eventData = Object.fromEntries(formData);
+
+    const responseEvent = await fetch("/api/events", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(eventData),
+    });
+
+    if (responseEvent.ok) {
+      const data = await responseEvent.json();
+      const responseKid = await fetch(`/api/kids/${kidData?._id}`, {
+        method: "PATCH",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          events: [...kidData?.events, data.data._id],
+        }),
+      });
+
+      if (responseKid.ok) {
+        mutate();
+      }
+    }
+  }
```

Comment:
This logic belongs to the respective page, in this case the `pages/[id].js`.
Anything that manages data and makes API calls goes to the page level and is passed down to the components and handle functions.

Here, you'd pass it down to the `KidProfile` first and then to the `EventForm` second. That way, you'll have no logic in your components anymore and that's exactly what you want. Components should be dump.